pkg/audit/stats_reporter.go

```diff
@@ -2,111 +2,102 @@ package audit
 
 import (
 	"context"
+	"errors"
 	"time"
 
-	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
-	"github.com/open-policy-agent/gatekeeper/v3/pkg/util"
-	"go.opencensus.io/stats"
-	"go.opencensus.io/stats/view"
-	"go.opencensus.io/tag"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
 )
 
 const (
 	violationsMetricName       = "violations"
 	auditDurationMetricName    = "audit_duration_seconds"
 	lastRunStartTimeMetricName = "audit_last_run_time"
 	lastRunEndTimeMetricName   = "audit_last_run_end_time"
+	enforcementActionKey       = "enforcement_action"
 )
 
 var (
-	violationsM       = stats.Int64(violationsMetricName, "Total number of audited violations", stats.UnitDimensionless)
-	auditDurationM    = stats.Float64(auditDurationMetricName, "Latency of audit operation in seconds", stats.UnitSeconds)
-	lastRunStartTimeM = stats.Float64(lastRunStartTimeMetricName, "Timestamp of last audit run starting time", stats.UnitSeconds)
-	lastRunEndTimeM   = stats.Float64(lastRunEndTimeMetricName, "Timestamp of last audit run ending time", stats.UnitSeconds)
-
-	enforcementActionKey = tag.MustNewKey("enforcement_action")
+	violationsM       metric.Int64ObservableGauge
+	auditDurationM    metric.Float64Histogram
+	lastRunStartTimeM metric.Float64ObservableGauge
+	lastRunEndTimeM   metric.Float64ObservableGauge
+	endTime           time.Time
```

Comment:
I think there are concurrency concerns with using global variables like this.

Hiding the state inside the reporter and appropriate uses of mutexes/atomic should help with this.

Reply:
Could you shed some light on why would we need mutex here? as I understand it there is only one audit process going on at a time right?